lib/trino-orc/src/main/java/io/trino/orc/OrcWriter.java

```diff
@@ -357,6 +355,15 @@ private void flushStripe(FlushReason flushReason)
         // write all data
         orcDataSink.write(outputData);
 
+        if (flushReason == CLOSED) {
```

Comment:
there is plenty of fields in both `OrcWriter` and `ParquetWriter`, (e.g `orcDataSink`). These are not freed by this code. Writers themselves seem are big objects so it doesn't make sense to keep them when they are not needed

Reply:
Do any of these fields hold substantial memory?

Reply:
Yes